### ISSUE TITLE
Critical vulnerability in socket.io-parser

### ISSUE THREAD

**Issue by jattasNI** [ni/nimble#801]
## 🧹 Tech Debt
`npm audit` started failing due to [a new critical vulnerability in socket.io-parser](https://github.com/advisories/GHSA-qm95-pgcg-qqfq). This blocks the main PR workflow.

The package is pulled in by `karma` and `@11y/eleventy`:
- eleventy has fixed [a similar issue](https://github.com/11ty/eleventy/issues/2213) in the past but [I haven't seen a discussion of this vulnerability yet](https://github.com/11ty/eleventy/issues?q=is%3Aissue+socket.io)
- karma has [an issue to track the vulnerability](https://github.com/karma-runner/karma/issues/3823) but no fix yet

```
npm ls socket.io-parser

├─┬ @ni/nimble-components@15.1.1 -> ./packages/nimble-components
│ └─┬ karma@6.3.4
│   └─┬ socket.io@3.1.2
│     └── socket.io-parser@4.0.4
└─┬ @ni/site@1.0.0-beta.0 -> ./packages/site
  └─┬ @11ty/eleventy@1.0.2
    └─┬ browser-sync@2.27.10
      ├─┬ browser-sync-ui@2.27.10
      │ └─┬ socket.io-client@4.5.2
      │   └── socket.io-parser@4.2.1
      └─┬ socket.io@4.5.2
        └── socket.io-parser@4.2.1
```

**Comment by rajsite** [comment]
@fredvisser does Snyk replace the functionality that `npm audit` provided? If so then the action for this issue is to remove the audit steps from `main.yml`. Marking as "discussion" to see if we can make this issue go away.

**Comment by fredvisser** [comment]
Snyk doesn't support monorepos by default, so isn't picking up any of the package specific package.json files. So the current setup isn't good enough.
While they have a CLI flag that will find the other package.json files, we're using a web hook which doesn't appear to expose the option. 

We could switch to a their Github Action or we could add the [Github Dependency Review Action](https://github.com/actions/dependency-review-action) which would keep the workflow entirely with Github.

**Comment by rajsite** [comment]
Fixed in #1930

**Comment by rajsite** [comment]
@fredvisser looking at the dependency review check instructions it says to check the following page: https://github.com/ni/nimble/security/dependabot

where there I see non-dismissed critical vulns, but it doesn't look like dependency review check is complaining about those, see build:
https://github.com/ni/nimble/actions/runs/8439038563

and log:
https://github.com/ni/nimble/actions/runs/8439038563/job/23112684298#step:3:8
